services/src/main/java/org/exoplatform/news/rest/NewsTargetsRestResourcesV1.java

```diff
@@ -0,0 +1,53 @@
+package org.exoplatform.news.rest;
+
+import io.swagger.annotations.*;
+import org.exoplatform.common.http.HTTPStatus;
+import org.exoplatform.services.log.ExoLogger;
+import org.exoplatform.services.log.Log;
+import org.exoplatform.services.rest.resource.ResourceContainer;
+import org.exoplatform.social.metadata.MetadataService;
+import org.exoplatform.social.metadata.model.Metadata;
+import org.exoplatform.social.metadata.model.MetadataKey;
+
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+import java.util.List;
+
+@Path("v1/newsTargets")
+@Api(tags = "v1/newsTargets", value = "v1/newsTargets", description = "Managing news targets")
+public class NewsTargetsRestResourcesV1 implements ResourceContainer {
+
+  private static final Log       LOG                             = ExoLogger.getLogger(NewsTargetsRestResourcesV1.class);
+
+  private static final String       NEWS_TARGET_TYPE                             = "newsTarget";
+
+  private MetadataService metadataService;
+
+  public NewsTargetsRestResourcesV1(MetadataService metadataService) {
+    this.metadataService = metadataService;
+  }
+
+  @Path("getTargetsByType/{type}")
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @ApiOperation(value = "Get all news targets by a giving type", httpMethod = "GET", response = Response.class, produces = "application/json")
+  @ApiResponses(value = { @ApiResponse(code = HTTPStatus.OK, message = "Request fulfilled"),
+      @ApiResponse(code = HTTPStatus.BAD_REQUEST, message = "Invalid query input"),
+      @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal server error"), })
+  public Response getTargetsByType(@Context UriInfo uriInfo,
+                         @ApiParam(value = "Target type", required = true) @PathParam("type") long type) {
+    try {
+      List<Metadata> metadata = metadataService.getMetadatasByType(type);
```

Comment:
here we should add the news targeting service layer to be called from the rest endpoint
the type will be fixed and specified in the service layer